02-microarray/clustering/clustering_example.Rmd

```diff
@@ -1,20 +1,97 @@
 ---
-title: "Clustering Data"
-author: "ALSF CCDL - Candace Savonen"
+title: "Clustering Heatmap - Microarray"
+author: "CCDL for ALSF"
 output:   
   html_notebook: 
     toc: true
     toc_float: true
 ---
 
-*Purpose*: This notebook illustrates one way that you can use harmonized data 
-from refine.bio in downstream analyses.
+# Purpose of the analysis:
 
-## 1) Install libraries
-This script uses the bioconductor R package ComplexHeatmap for clustering and 
-creating a heatmap.
-<br/> <i> Citation:</i> Gu Z, Eils R, Schlesner M (2016). “Complex heatmaps 
-reveal patterns and correlations in multidimensional genomic data.” Bioinformatics.
+Clustering can help you investigate which samples (or genes) are more related to each other.
+Here we use the Bioconductor R package `ComplexHeatmap` for clustering and creating a heatmap.
+In this example, the gene expression matrix is subset to only include genes with high variance.
+This set of genes is then used to create a heatmap.
+
+## Further learning resources about this analysis
+
+ - [Clustering strategies](https://www.statsandr.com/blog/clustering-analysis-k-means-and-hierarchical-clustering-by-hand-and-in-r/)  
+ - [Complex Heatmap book](https://jokergoo.github.io/ComplexHeatmap-reference/book/)  
+ - [Complex Heatmap vignette](https://bioconductor.riken.jp/packages/3.4/bioc/vignettes/ComplexHeatmap/inst/doc/s1.introduction.html)  
+ - [Complex Heatmap publication](https://academic.oup.com/bioinformatics/article/32/18/2847/1743594)  
+ 
+***
+
+# How to run this example
+
+## About dataset we are using for this example
+
+For this example analysis, we will use this [zebrafish microarray dataset](
+https://www.refine.bio/experiments/GSE12955/gene-expression-profile-of-zebrafish-kidney-side-population-sp-cells)
+This dataset has two experimental groups of lymphoid cells from zebrafish kidney.
+Using Hoechst fluorescence intensity, these were subdivided into the SP and MP populations. 
+
+## Obtaining the dataset from refinebio
+
+Go to this [dataset's page on refinebio](https://www.refine.bio/experiments/GSE12955/gene-expression-profile-of-zebrafish-kidney-side-population-sp-cells).  
+
+Click the "Download Now" button on the right side of this screen.  
+
+<img src="screenshots/download-now.png" width=200>  
+
+Fill out the pop up window with your email and our Terms and Conditions:  
+
+<img src="screenshots/download-email.png" width=500>  
+
+This will download a zip file that has a name with a series of letters and numbers and ends in `.zip`. 
+Double clicking should unzip this for you and create a folder of the same name.  
```

Comment:
Is the idea to rename this folder as `data`? This is what I gather from the `Set up your analysis` section below.

If so, does this mean that we are going to get rid of the unzipping of files step in R?

Also, are we going to have them place all of their files in the `data` folder instead of having them separate their data by experiment, which has been automatically encouraged in the refinebio-examples notebooks each time the users unzip the file and a separate folder specific to that experiment is created within the `data` folder.

Reply:
These are all excellent questions that I'm not 100% sure what the best plan is.

 In this example I'm *trying* to tell people to just unzip the download folder by double clicking since we can't know what the download zip file name will be. Then I'm trying to tell them to move the files they need into a new folder they create which includes a `data` folder. My thought is we can still have the plots and results folders created in the Rmd. 

^If what I have written in the steps now doesn't reflect this (or what I'm describing here is unclear), let me know and I'll try to change it to be more clear.

Reply:
The screenshot at line 74 under the `Set up your analysis folder` section appears to have the files within that unzipped folder in the `data` directory. Your steps are clear but that would meant that folder structure will be `data/GSE1235.tsv` instead of `data/GSE1235/GSE1235.tsv` which can get confusing if they are following other notebooks and downloading additional datasets. 

Is that what we want however, for everything to just live in the main data folder instead in subdirectories of the data folder?

Reply:
I think just having one level of data folder makes sense. I have them copy over the files into that new folder instead of making the download folder into the data folder. Does this make sense?